deepspeed/runtime/zero/stage2.py

```diff
@@ -300,7 +306,8 @@ def __init__(self,
             # each process will compute on a different part of the partition
             data_parallel_partitions = self.get_data_parallel_partitions(
                 self.fp16_groups_flat[i],
-                i)
+                i,
+                self.nccl_start_alignment_factor)
```

Comment:
@amathews-amd since you have already aligned self.fp16_groups_flat, do we need this alignment on 310 again? We should be able to leave the get_data_parallel_parititons method unmodified  no?